hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/WritableECContainerProvider.java

```diff
@@ -203,8 +203,14 @@ private ContainerInfo allocateContainer(ReplicationConfig repConfig,
 
     Pipeline newPipeline = pipelineManager.createPipeline(repConfig,
         excludedNodes, Collections.emptyList());
+    // the returned ContainerInfo should not be null (due to not enough space in the Datanodes specifically) because
```

Comment:
Could newPipeline be null when DN is close to full?

Reply:
Good point. I checked, pipeline will not be null. There's a `checkPipeline(pipeline)` method that checks for null and throws an exception instead.